.eslintrc.js

```diff
@@ -43,6 +43,10 @@ module.exports = {
       }
     }],
     'one-var': ['error', 'never'],
+    'import/no-extraneous-dependencies': ['error', {
+      /* the following folders can import devDependencies */
+      devDependencies: ['test/**', 'tests/**', 'src/debug/**', 'examples/**'],
```

Comment:
I think I agree. Let me rephrase to be extra sure:

1. Either debug mode is not available *at all* for client app, and then it is not in `src/debug` but in `debug/` for example and `debug/` is added to `npmignore`
1. Or it is available to client app, and in this case, its dependencies are in `optionalDependencies`.

And you prefer option 2., is that right?